### DISCUSSION TITLE
Combine multiple PartitionsDefinitions into one PartitionsDefinition

### DISCUSSION THREAD

**Discussion by weberdavid** [dagster-io/dagster#19773] [Q&A]
We are currently building the orchestration for ETL pipelines with Dagster.
For each data entity that we process, we set a DynamicPartitionsDefinition. Each DynamicPartition in this definition represents the string-timestamp of the file / partition that we process. So we receive multiple file deliveries for each data entity every week.

Later on, we use dbt to build more sophisticated tables, such as joining multiple data entities together (e.g. building a DWH).
We still want to partition the DWH-dbt assets, which results in a specific requirement:

Due to joining data entities in this dbt model, we would have the different DynamicPartitionsDefinitions of the data entities coming together in one Dagster asset. How could this be solved?

Our current approach would be to make a new DynamicPartitionsDefinition for this asset, but then we lose the full partition lineage of this entity.

How could this be adressed? Are there best practices how to handle these cases? From our perspective, it seems to be a quite generic requirement that different assets may merge into one later in a pipeline, which results in this requirement of also combing their partitions.

**Comment by sryza** [comment]
When you say "combine" in this case, is the idea that you would want to combined partitions definition to include the union of the values in the upstream partitions definitions?

Alas there is not currently a great way to do this.  If this is what you're looking for, I'd recommend filing a Github issue with this request.

**Reply by weberdavid** [reply]
Yes, producing the union of upstream partitions is exactly what I am looking for. In the best case even with the possibility to see, which partitions are coming from which PartitionsDefinition.

I will file an issue, thanks for the prompt reply @sryza!

**Comment by weberdavid** [comment]
Will close due to corresponding issue #19789
